src/main/java/com/snowflake/kafka/connector/internal/streaming/SnowpipeStreamingFileType.java

```diff
@@ -0,0 +1,76 @@
+package com.snowflake.kafka.connector.internal.streaming;
+
+import java.util.Locale;
+import net.snowflake.ingest.utils.Constants;
+import org.apache.kafka.common.config.ConfigDef;
+
+/**
+ * Enum representing the allowed values for config {@link
+ * com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig#SNOWPIPE_STREAMING_FILE_TYPE}
+ *
+ * <p>Note: Do not use ARROW format unless absolutely needed, contact the support team if needed be.
+ */
+public enum SnowpipeStreamingFileType {
+
+  /* Arrow file format corresponds to bdec version 1, use only as a fallback when PARQUET doesnt work */
+  ARROW(Constants.BdecVersion.ONE),
+
+  /* Default BDEC file format PARQUET */
+  PARQUET(Constants.BdecVersion.THREE),
+  ;
+
+  /** Bdec version defined in Ingest SDK */
+  private final Constants.BdecVersion bdecVersion;
+
+  /**
+   * @return the bdec version associated with the file type. BDEC version later is passed in
+   *     Streaming Client generation.
+   */
+  public Constants.BdecVersion getBdecVersion() {
+    return bdecVersion;
+  }
+
+  /**
+   * Validator to validate {@link
+   * com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig#SNOWPIPE_STREAMING_FILE_TYPE}
+   */
+  public static final ConfigDef.Validator VALIDATOR =
```

Comment:
can we add code coverage for this? feels weiiirdd to add validation for a validator, but i'd rather a test fail than the entire test suite fail on this validator if the file type ever changes under us